Forwarded: https://github.com/MariaDB/server/pull/2482
Origin: https://patch-diff.githubusercontent.com/raw/MariaDB/server/pull/2482.patch
From: Daniel Black <daniel@mariadb.org>
Date: Mon, 6 Feb 2023 18:00:15 +1100
Subject: [PATCH] MDEV-30572: my_large_malloc will only retry on ENOMEM

Correct error in to only say "continuing to smaller size" if it really
is.

This is designed to fix https://bugs.debian.org/1006529
---
 mysys/my_largepage.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/mysys/my_largepage.c
+++ b/mysys/my_largepage.c
@@ -376,7 +376,7 @@ uchar *my_large_malloc(size_t *size, myf
       ptr= NULL;
       if (my_flags & MY_WME)
       {
-        if (large_page_size)
+        if (large_page_size && errno == ENOMEM)
         {
           my_printf_error(EE_OUTOFMEMORY,
                           "Couldn't allocate %zu bytes (Large/HugeTLB memory "
